src/authentication/githubServer.ts

```diff
@@ -2,36 +2,11 @@ import * as https from 'https';
 import * as vscode from 'vscode';
 import Logger from '../common/logger';
 import { agent } from '../common/net';
-import { handler as uriHandler } from '../common/uri';
-import { PromiseAdapter, promiseFromEvent } from '../common/utils';
-import { HostHelper, IHostConfiguration } from './configuration';
-import { listHosts, onDidChange as onKeychainDidChange, toCanonical } from './keychain';
-import uuid = require('uuid');
-import { EXTENSION_ID } from '../constants';
-
-const SCOPES: string = 'read:user user:email repo write:discussion';
-const GHE_OPTIONAL_SCOPES: { [key: string]: boolean } = { 'write:discussion': true };
-
-const AUTH_RELAY_SERVER = 'vscode-auth.github.com';
+import { HostHelper } from './configuration';
 
 export class GitHubManager {
 	private _servers: Map<string, boolean> = new Map().set('github.com', true);
 
-	private static GitHubScopesTable: { [key: string]: string[] } = {
```

Comment:
There are a lot more scopes removed here than are added in github/credentials.ts. Are they just not needed?

Reply:
Yeah, these were equivalences between different scopes, used to validate that a token had the correct permissions. Since the auth provider should always return a token with the exact scopes we requested, this isn't necessary. (Before, a user could provide their own token, and the GHE token also had a slightly different list of scopes)